src/App.jsx

```diff
@@ -14,45 +16,104 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const FILTER_ALPHABETICALLY = 'Sort alphabetically';
+const FILTER_LENGTH = 'Sort by length';
+
+export const App = () => {
+  const [chosenFilter, setChosenFilter] = useState('');
+  const [reverseBtn, setReverseBtn] = useState(false);
+  let sortedGoods = [...goodsFromServer];
+
+  if (chosenFilter) {
+    switch (chosenFilter) {
+      case FILTER_ALPHABETICALLY:
+        sortedGoods = sortedGoods.sort((good1, good2) => {
+          if (reverseBtn) {
+            return good2.localeCompare(good1);
+          }
+
+          return good1.localeCompare(good2);
+        });
+        break;
+
+      case FILTER_LENGTH:
+        sortedGoods = sortedGoods.sort((good1, good2) => {
+          if (reverseBtn) {
+            if (good2.length !== good1.length) {
+              return good2.length - good1.length;
+            }
+
+            return good2.localeCompare(good1);
+          }
+
+          return good1.length - good2.length;
+        });
+        break;
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+      default:
+        return 0;
```

Comment:
you can return 0 inside `sort` callback function - it'd be a valid value
here just do `break;` - no need to return anything